File: web/src/render/htmlSpanRenderer.ts
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";

const renderer: Renderer = {
	name: "html_span",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	state.output += node.content;
}

File: dotnet/Allmark/Render/HtmlSpanRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class HtmlSpanRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "html_span",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        state.Output.Append(node.Content);
    }
}
